src/shell/layer.rs

```diff
@@ -0,0 +1,110 @@
+/// WLR layer abstraction
```

Comment:
Note that module doc comments are `//!`. You can always run `cargo doc` to check if the docs appear in the right spot.

This would also be a great place to summarize the scope of the layer module and provide a code example. The code example is probably best derived from the examples you've written.